### PR TITLE
Unit of work

### PR DESCRIPTION
**Pull request by TordJoranger**

### CODE REVIEW

File: src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs [hunk 1]
```diff
@@ -1,10 +1,12 @@
 ﻿using Microsoft.EntityFrameworkCore;
+using QueueReceiver.Core.Interfaces;
 using QueueReceiver.Core.Models;
+using System.Threading.Tasks;
 using Person = QueueReceiver.Core.Models.Person;
```
Comment (paal123) [inline src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs]: This is never a good sign... Why do we need to explicitly set using Person?

File: src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 
 namespace QueueReceiver.Infrastructure.Data
 {
-    public class ApplicationDbContext : DbContext
+    public class ApplicationDbContext : DbContext, IUnitOfWork
```
Comment (paal123) [inline src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs]: Rename to QueueReceiverServiceContext to align with Preservation naming standard

File: src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs [hunk 3]
```diff
@@ -1,10 +1,12 @@
 ﻿using Microsoft.EntityFrameworkCore;
+using QueueReceiver.Core.Interfaces;
 using QueueReceiver.Core.Models;
+using System.Threading.Tasks;
 using Person = QueueReceiver.Core.Models.Person;
 
 namespace QueueReceiver.Infrastructure.Data
```
Comment (paal123) [inline src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs]: Move to QueueReceiver.Infrastructure to align with Preservation setup

**Review by paal123 — CHANGES_REQUESTED** [review]
Rename namespace QueueReceiver.Infrastructure.Data to QueueReceiver.Infrastructure.EntityConfiguration

And

Core.Models should be in a domain project instead

File: src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs (hunk 1 above)
Comment (llykkelarsen) [reply src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs]: Pretty sure this is my bad. It is not needed and may be removed.

**Comment by TordJoranger** [comment]
> Rename namespace QueueReceiver.Infrastructure.Data to QueueReceiver.Infrastructure.EntityConfiguration
> 
> And
> 
> Core.Models should be in a domain project instead

So a new project that only contains the models? and that doesn't depend on anything then?

**Pull request merged**